keyboards/gboards/gergoplex/matrix.c

```diff
@@ -0,0 +1,241 @@
+/*
+
+Copyright 2013 Oleg Kostyuk <cub.uanic@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "matrix.h"
```

Comment:
This should be converted to matrix lite.

Check the ergodox ez's matrix for reference.

Reply:
Done so in the last commit, but the keyboard just stopped working.
Can you please give a cue what did I do wrong, @drashna ?

Reply:
Hmm.  My guess is that some of the config isn't correct. 

It's just the expander side that stopped working?

Reply:
In `static matrix_row_t read_cols(uint8_t row) {` @germ originally did a write before reading i2c, it looks like that's missing, maybe thats it? (I'm new to qmk_firmware, so not sure)

https://github.com/germ/qmk_firmware/blob/39c86e080dc04b68ee08387dcb5144c88cb44855/keyboards/gboards/k/gergoplex/matrix.c#L223-L224

Reply:
> It's just the expander side that stopped working?

Both, actually. Is this a clue?

Reply:
I'm sorry, I lost track of this. 

And yeah, it means that the matrix isn't being processed correctly.  Give me a couple of days, and I'll see if I can fix up some code. 

Unfortunately, I don't have a gergoplex to test on, and I'm not sure how close the other boards are in design, so testing is ... problematic.  

If it's more than a few days, give me a ping on the discord.

Reply:
Thanks. I might have some time this weekend to debug it. On the one hand (pun intended) it's confusing that both parts of the keyboard stop working, even though one of the matrices is scanned via I2C, and another with matrix lite's helper code.
The other thing that confuses me is that rows and columns are mixed up.
A good thing is that if you checkout `@~1`, you have a working code.

Reply:
Yeah, that design is pulled from the ergodox (og), and ... IIRC it's to help with scan speeds